src/main/java/mj/chess/Board.java

```diff
@@ -21,4 +39,31 @@ public int size() {
     public Pawn findPawn(int index) {
         return pawns.get(index);
     }
+
+    public void initialize() {
+        initPawns(Color.WHITE, Pawn.WHITE_REPRESENTATION, SECOND_ROW, rowOfWhitePawns);
+        initPawns(Color.BLACK, Pawn.BLACK_REPRESENTATION, SEVENTH_ROW, rowOfBlackPawns);
+    }
+
+    private void initPawns(Color color, char representation, final int NTH_ROW, StringBuilder rowOfPawns) {
+        for (int i = 0; i < MAX_SIZE; i++) {
+            pawns.add(new Pawn(color));
+            rowOfPawns.append(representation);
+        }
+        cellsOfBoard[NTH_ROW] = (color == Color.BLACK) ? getRowOfBlackPawns() : getRowOfWhitePawns();
+    }
```

Comment:
리팩토링을 해보자면.. 

```suggestion
    public void initialize() {
        initPawns(Color.WHITE);
        initPawns(Color.BLACK);
        initCellsOfBoard();
    }

    private void initPawns(Color color, int nthRow) {
        for (int i = 0; i < MAX_SIZE; i++) {
            pawns.add(new Pawn(color));
        }
    }
    
    private void initCellsOfBoard() {
      cellsOfBoard[WHITE_PAWN_ROW] = pawns.stream().filter(p -> p.isColor(Color.WHITE)).map(p -> p.getRepre()).joinString() 
      cellsOfBoard[BLACK_PAWN_ROW] = pawns.stream().filter(p -> p.isColor(Color.BLACK)).map(p -> p.getRepre()).joinString()
    }
```

코드로 직접 리뷰를 드리기보다 생각할 것을 던져드리려고 했는데.. 쓰다보니 코드를 적어버렸네요 ㅎㅎ
제가 하고 싶은 말은 initPawns메서드가 일단 두가지 일을 하고 있는 것으로 보였구요,

1. pawns을 넣는 부분
2.cells 초기화 하는 부분

기능 분리를 좀 해봤고.. (여기서 힌트는 메서드 인자가 3,4개 이상 넘어가고 있다면 어딘지 쓰이는 곳이 많다는 사인일 수도 있습니다.) 

그리고 initPawns 메서드에서 color가 지정된 Pawn의 getRepresentation() 메서드가 있음에도 initPawns에 char 를 인자로 넘겨서 pawn 객체를 충분히 활용하고 있지 못한 점이 아쉽네요 ㅠ 

제가 예시로 코드를 드렸는데 한번 살펴봐 주시고, 어떤 장,단점이 있을지 의견을 주시면 좋을것 같습니다.!

(코드는 좀 의사(psuedo)코드라 정확하지 않습니다. 정답도 아니구요!)

Reply:
추가 코멘트 드리면 파라미터 컨벤션을 지켜주세요!

Reply:
본 리뷰에서 Larry가 강조하는 건 코드의 재사용성으로 이해했습니다. 코드의 재사용성을 높이는 것의 장점을 세 가지로 정리했습니다. 
1. 성능향상, JVM의 Execution Engine 중 JIT Compiler는 한 번 읽어드린 코드에 대해서는 line by line이 아니라 한 번에 실행합니다. 따라서 재사용되는 코드가 많을 수록 프로그램 전체의 성능은 향상됩니다.
2. 수정작업 시 효율성 증가, 코드의 재사용률이 높으면 수정사항이 발생해도 개발자가 일일이 변경할 코드의 양은 줄어들 것입니다.
3. 에러 발생 감소, 사람이 개발하는 이상 인간적인 실수가 발생할 수 있습니다. 사람이 일일이 건드는 변수의 양을 줄이고 반복적으로 사용하는 코드가 많아진다면 인간적인 실수도 함께 줄 것입니다.

단점은 아무래도 재사용을 하기 위해 다른 클래스의 코드를 가져와야 하기 때문에 코드의 간결성이 떨어진다는 것으로 볼 수 있습니다. 하지만 이것도 어떻게 재사용하는지에 따라서 오히려 간결해질 수 있습니다.